test/unit/region.spec.js

```diff
@@ -593,9 +593,19 @@ describe('region', function() {
       expect(view1.destroy).to.have.been.called;
     });
 
+    it('should call "empty" even if a new view is attached to the DOM', function() {
+
+      this.sinon.spy(region, 'empty');
+      const preRenderedView = new View({ el: '#pre-rendered', template: _.noop});
```

Comment:
if i remove template then it fails because region tries to render templateless view
 1028 passing (4s)
  1 failing
1) region when a view is already shown and showing another should call "empty" even if a new view is attached to the DOM:
     TypeError: template is not a function

Reply:
This is because the view is not considered rendered because there's no content rendered in the `el`.  Put something in the `el` and `preRenderedView.isRendered()` will be true prior to showing it in the region and the `template` won't be necessary.